server/athenian/api/internal/miners/jira/issue.py

```diff
@@ -812,74 +816,150 @@ def _validate_and_clean_issues(df: pd.DataFrame, acc_id: int) -> pd.DataFrame:
+    """Mapped JIRA information for a pull request."""
+
+    ids: list[str]
+    """JIRA issue identifiers mapped the PR."""
+    projects: list[str]
+    """JIRA issue projects mapped to the PR."""
+    priorities: list[str]
+    """JIRA issue priorities mapped to the PR."""
+    types: list[str]
+    """JIRA issue priorities types to the PR."""
+
+    def merge(self, other_mapping: PRJIRAMapping) -> PRJIRAMapping:
+        """Return a new mapping with merged also the info from another mapping."""
+        return self.__class__(
+            sorted(set(self.ids).union(other_mapping.ids)),
+            sorted(set(self.projects).union(other_mapping.projects)),
+            sorted(set(self.priorities).union(other_mapping.priorities)),
+            sorted(set(self.types).union(other_mapping.types)),
+        )
+
+
 class PullRequestJiraMapper:
     """Mapper of pull requests to JIRA tickets."""
 
-    @classmethod
-    async def append_pr_jira_mapping(
-        cls,
-        prs: PullRequestFactsMap,
-        meta_ids: Tuple[int, ...],
-        mdb: DatabaseLike,
-    ) -> None:
-        """Load and insert "jira_id" to the PR facts."""
-        pr_node_ids = defaultdict(list)
-        for node_id, repo in prs:
-            pr_node_ids[node_id].append(repo)
-        jira_map = await cls.load_pr_jira_mapping(pr_node_ids, meta_ids, mdb)
-        for pr_node_id, jira in jira_map.items():
-            for repo in pr_node_ids[pr_node_id]:
-                try:
-                    prs[(pr_node_id, repo)].jira_ids = jira
-                except KeyError:
-                    # we removed this PR in JIRA filter
-                    continue
-
     @classmethod
     @sentry_span
-    async def load_pr_jira_mapping(
+    async def load(
         cls,
         prs: Collection[int],
-        meta_ids: Tuple[int, ...],
+        meta_ids: tuple[int, ...],
         mdb: DatabaseLike,
-    ) -> Dict[int, List[str]]:
-        """Fetch the mapping from PR node IDs to JIRA issue IDs."""
+    ) -> dict[int, PRJIRAMapping]:
+        """Load the JIRA mapping for each PR in `prs`."""
         nprji = NodePullRequestJiraIssues
+
         if len(prs) >= 100:
             node_id_cond = nprji.node_id.in_any_values(prs)
         else:
             node_id_cond = nprji.node_id.in_(prs)
-        rows = await mdb.fetch_all(
-            sql.select([nprji.node_id, Issue.key])
-            .select_from(
-                sql.outerjoin(
-                    nprji,
-                    Issue,
-                    sql.and_(
-                        nprji.jira_acc == Issue.acc_id,
-                        nprji.jira_id == Issue.id,
-                    ),
-                ),
-            )
-            .where(
-                sql.and_(
-                    node_id_cond,
-                    nprji.node_acc.in_(meta_ids),
-                ),
-            ),
+        cols = [nprji.node_id, Issue.key, Issue.priority_name, Issue.type, Issue.project_id]
+        from_clause = sa.outerjoin(
```

Comment:
question: outerjoin here is maybe not needed ? we have a FK (nprji.jira_acc, nprji.jira_id) => (Issue.acc_id, Issue.id)

Reply:
As a rule of thumb, don't change queries because there is a high chance that they were optimized for performance. In this particular case, see "The hidden advantage of left join" in https://towardsdatascience.com/how-we-optimized-postgresql-queries-100x-ff52555eabe